File: backend/src/main/java/com/taskmaster/domain/model/Task.java
package com.taskmaster.domain.model;

import lombok.Builder;
import lombok.Data;
import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
public class Task {
    private UUID id;
    private UUID userId;
    private String title;
    private String description;
    private String priority;
    private String category;
    private boolean routine;
    private OffsetDateTime dueDate;
    private String status;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: frontend/src/types/task.ts
export interface Task {
  id: string;
  title: string;
  description?: string;
  priority: 'High' | 'Medium' | 'Low';
  category?: string;
  routine: boolean;
  dueDate?: string;
  status: string;
  createdAt: string;
}

export interface TaskRequest {
  title: string;
  description?: string;
  priority: 'High' | 'Medium' | 'Low';
  category?: string;
  routine: boolean;
  dueDate?: string;
  userId?: string;
}

export interface TaskResponse {
  id: string;
  title: string;
  description?: string;
  priority: 'High' | 'Medium' | 'Low';
  category?: string;
  routine: boolean;
  dueDate?: string;
  status: string;
  createdAt: string;
}
